cmake/RunReverseDD.cmake

```diff
@@ -0,0 +1,201 @@
+# These need to be defined by the caller
+# SOURCE_DIR
+# BINARY_DIR
+# ENERGYPLUS_EXE
+# IDF_FILE
+# BUILD_FORTRAN
+# TEST_FILE_FOLDER
+# ENERGYPLUS_FLAGS
+# PYTHON_EXECUTABLE
+
+get_filename_component(IDF_NAME "${IDF_FILE}" NAME_WE)
+get_filename_component(IDF_EXT "${IDF_FILE}" EXT)
+get_filename_component(EXE_PATH "${ENERGYPLUS_EXE}" PATH)
+
+# Create path variables
+set(OUTPUT_DIR_NAME "testfiles_ReverseDD")
+set(IDF_PATH "${SOURCE_DIR}/${TEST_FILE_FOLDER}/${IDF_FILE}")
+set(PRODUCT_PATH "${BINARY_DIR}/Products/")
+set(EXE_PATH "${EXE_PATH}/")
+set(EPW_PATH "${SOURCE_DIR}/weather/${EPW_FILE}")
+
+# Copy IDD to Executable directory if it is not already there
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${PRODUCT_PATH}/Energy+.idd" "${EXE_PATH}")
+
+# Delete the root test directory which will delete any previous Regular/Reversed subdirectories
+execute_process(COMMAND "${CMAKE_COMMAND}" -E remove_directory "${BINARY_DIR}/${OUTPUT_DIR_NAME}/${IDF_NAME}")
+
+# list zipping is available as of 3.17, which is our minimum version, :+1:
+set(MODE Regular Reversed)
+set(VAR "" Y)
+
+message("Running reverse design day test for file ${IDF_NAME}")
+
+foreach(M V IN ZIP_LISTS MODE VAR)
+
+  set(OUTPUT_DIR_PATH "${BINARY_DIR}/${OUTPUT_DIR_NAME}/${IDF_NAME}/${M}/")
+
+  # Create a new directory for output
+  execute_process(COMMAND "${CMAKE_COMMAND}" -E make_directory "${OUTPUT_DIR_PATH}")
+
+  # Read the file contents to check for special cases
+  file(READ "${IDF_PATH}" IDF_CONTENT)
+
+  # Convert flags back to CMake list
+  string(STRIP ${ENERGYPLUS_FLAGS} ENERGYPLUS_FLAGS)
+  string(REPLACE " " ";" ENERGYPLUS_FLAGS_LIST ${ENERGYPLUS_FLAGS})
+
+  # Use EPMacro if necessary
+  list(FIND ENERGYPLUS_FLAGS_LIST -m EPMACRO_RESULT)
+
+  if("${EPMACRO_RESULT}" GREATER -1)
```

Comment:
A chunk of this should be DRY'ed up because it is being reused from in the `RunSimulation.cmake` file...but that said, that file has had one meaningful change in the last 6 years, so it's not my highest priority.